library/blob.c

```diff
@@ -330,10 +330,12 @@ inline static uint64_t eblob_validate_ctl_flags(struct eblob_backend *b, uint64_
 	if (b->cfg.blob_flags & EBLOB_NO_FOOTER)
 		flags |= BLOB_DISK_CTL_NOCSUM;
 
-	if (!(flags & BLOB_DISK_CTL_NOCSUM))
-		flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
-	else
-		flags &= ~BLOB_DISK_CTL_CHUNKED_CSUM;
+	/*
+	 * We have to add BLOB_DISK_CTL_CHUNKED_CSUM because it shows
+	 * that footer was prepared with specific size, even if the record
+	 * has BLOB_DISK_CTL_NOCSUM flag.
+	 */
+	flags |= BLOB_DISK_CTL_CHUNKED_CSUM;
```

Comment:
It is needed because even if a record wasn't checksummed (because it was written with `BLOB_DISK_CTL_NOCSUM`), it still has footer whose size is different for old and for new data.

Flag `BLOB_DISK_CTL_CHUNKED_CSUM` shows that footer has specific structure. 
Flag `BLOB_DISK_CTL_NOCSUM` shows that checksums from the footer is unfilled and should be ignored.

We always set flag `BLOB_DISK_CTL_CHUNKED_CSUM` to new record's flag because we do not write records with old checksums anymore.